src/GiantBranch.cpp

```diff
@@ -1703,7 +1706,11 @@ STELLAR_TYPE GiantBranch::ResolveElectronCaptureSN() {
     }
     else {                                                                                  // -no, treat as ONeWD 
         
-        m_Mass       = m_COCoreMass;                                                        
+        m_Mass       = std::min(m_COCoreMass,MCH);                                          // no WD masses above Chandrasekhar mass
```

Comment:
This change seems to me like a bandaid fix. If we're getting core masses above MCH, they should be undergoing SN, so they should not end up here anyway. I would suggest throwing an error here if the mass is too high, instead of resetting it to the lower MCH value.

Reply:
We don't throw errors.  We could display a warning or error message that the coremass was reset, but we can't just then return from the function - the show must go on, so we have to return with values that are sensible enough to continue the evolution.

One day we'll figure out how we want to respond to errors/events/situations like this (e.g. we might set an error and bubble it all the way back to the Evolve() function and terminate the evolution there with the appropriate error), but right now our strategy is to just plough on regardless...

Reply:
Fair point. I'd be happy if we kept the new change but added an Error message.

Reply:
That was an attempt at defensive programming -- but I accept the criticism, will fix. :)

Reply:
See fix.